src/kv_client/request.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::{
+    kv_client::{HasError, KvClient, RpcFnType, Store},
+    pd::PdClient,
+    Result,
+};
+
+use futures::future::BoxFuture;
+use futures::prelude::*;
+use futures::stream::BoxStream;
+use grpcio::CallOption;
+use kvproto::kvrpcpb;
+use std::sync::Arc;
+
+pub trait KvRequest: Sync + Send + 'static + Sized + Clone {
+    type Result;
+    type RpcRequest;
+    type RpcResponse: HasError + Clone + Send + 'static;
+    /// A single `KvRequest` can be divided into a number of RPC requests because the keys span
+    /// several regions or a single RPC request is too large. Most of the fields in these requests
+    /// share the same content while `Payload` is the part which differs among the requests.
+    type Payload;
```

Comment:
I don't think `Payload` is a great name here - it is very generic and to me feels more like a value-like thing than a key-like thing. It is maybe better than `KeyType`, but I think we could still do better.

Reply:
I change it to `Entries`. Any better suggestions?

Reply:
What are the possible instantiations? Key, Vec<Key>, BoundRange, and Vec<BoundRange>, are there any more?

Reply:
`Vec<kvrpcpb::Mutations>`

Reply:
Operand? KeyData? InputData? Data? (in decreasing order of preference)

Reply:
It's changed to `KeyData`.